src/values/AbstractObjectValue.js

```diff
@@ -510,10 +510,7 @@ export default class AbstractObjectValue extends AbstractValue {
             invariant(typeof P === "string");
             return memberExpressionHelper(o, P);
           },
-          {
-            skipInvariant: true,
-            isPure: true,
-          }
+          { skipInvariant: true }
```

Comment:
@trueadm I noticed that some property accesses were not emitted in my `NestedConditions` test and found your discussion in:

https://github.com/facebook/prepack/pull/2097/files/225590fe2e7c4e1dd40a0ea4b3d12d790900e609#r194060187

Getters could have a “pure” side-effect like `console.log`, correct? If a function call cannot be removed in pure scope, then a getter probably shouldn’t be as well?

This change is only tangentially related. I thought my serializer tests would snapshot the serialized output, but they don’t. Happy to remove if you disagree or move to another PR.

Here’s a gist with the serialized output without this “fix”:

https://gist.github.com/calebmer/3ba090cda321ce67576c8d3c73ec5670

Reply:
`console.log` is technically a side-effectful thing right? Why were you using getters for in your test? Minifiers like google closure compiler will also strip these too, so we added the same assumptions to our wiki doc. Let's talk about this in a separate issue – I don't want to side track the work you've done in this PR :)

Reply:
Thanks for the context. I thought this was a bug in my code 👍